src/Protocols/Wishbone.hs

```diff
@@ -0,0 +1,190 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fconstraint-solver-iterations=0 #-}
```

Comment:
This is dangerous, as it can make the type checker get stuck in an infinite loop. This is fairly tedious to debug if it ever happens. Please change it to some sensible limit.